src/app/FakeLib/TargetHelper.fs

```diff
@@ -322,36 +324,114 @@ let listTargets() =
 // Instead of the target can be used the list dependencies graph parameter.
 let doesTargetMeansListTargets target = target = "--listTargets"  || target = "-lt"
 
+/// Determines a parallel build order for the given set of targets
+let determineBuildOrder (d : TargetTemplate<unit>) =
```

Comment:
The problem I'm having is that generators for arbitrary depdendency graphs do not compose with the global mutable state (namely TargetDict).
Nevertheless it would be cool to have arbitrary dependency graphs for testing...

The only solution I came up with is having a different dependency graph and augmenting determineBuildOrder with functions for extracting dependencies..

Any thoughts on that?